pkg/server/storage/mysql.go

```diff
@@ -55,6 +55,40 @@ func (my *MysqlStorage) QueryHostkeys() (ResultIterator, error) {
 	return &SqlResultIterator{Rows: rows}, nil
 }
 
+func (my *MysqlStorage) RecordGitHubMapping(mapping map[string]string) error {
+	// Prepare for batch insert
+	entries := make([]string, 0, len(mapping))
+	values := make([]interface{}, 0, len(mapping)*2)
+	for ssoIdentity, githubUser := range mapping {
+		// Create one set of values for each mapping
+		entries = append(entries, "(?, ?)")
+		// Append matching values for mapping
+		values = append(values, ssoIdentity)
+		values = append(values, githubUser)
+	}
+	// Create prepared statement with necessary number of (?, ?) values
+	stmt := fmt.Sprintf(
+		"INSERT INTO github_user_mappings (sso_identity, github_username) VALUES %s ON DUPLICATE KEY UPDATE github_username=VALUES(github_username)",
```

Comment:
Worth noting that, this method (specifically using `VALUES(github_username)`) is deprecated as of MySQL 8.0.20. See https://dev.mysql.com/doc/refman/8.0/en/insert-on-duplicate.html for more details.